string time_stamp
int64 clock
